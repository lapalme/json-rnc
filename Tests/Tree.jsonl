null
3
[]
[3,4]
[3,[4,null],true]
[["a",["b",["c","d","e"]]]]
[false,[4.5,-456.789]]
{"c":null}
{"b":[{"c":3.45},"OK",true]}
{"a":23,"b":[1,true]}
