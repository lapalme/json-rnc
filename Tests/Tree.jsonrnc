##  example of a recursive type definition

start = list | tree | value
value = string | integer | number | boolean | null | list | tree  
list  = [value]
tree  = {*:value}@(minProperties=1,maxProperties=1)
