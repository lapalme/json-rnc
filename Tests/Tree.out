10:{u'a': 23, u'b': [1, True]}
{u'a': 23, u'b': [1, True]} does not match any alternative:
 -	array expected:	{u'a': 23, u'b': [1, True]}
 -	object length greater than 1	{u'a': 23, u'b': [1, True]}
 -{u'a': 23, u'b': [1, True]} does not match any alternative:
 -	string expected:	{u'a': 23, u'b': [1, True]}
 -	integer expected:	{u'a': 23, u'b': [1, True]}
 -	number expected:	{u'a': 23, u'b': [1, True]}
 -	boolean expected:	{u'a': 23, u'b': [1, True]}
 -	null expected:	{u'a': 23, u'b': [1, True]}
 -	array expected:	{u'a': 23, u'b': [1, True]}
 -	object length greater than 1	{u'a': 23, u'b': [1, True]}
10 objects read: 1 invalid, 0 bad, 0 with duplicate fields
Error Statistics
              1	{u'a': 23, u'b': [1, True]} does not match any alternative:
